src/compiler/checker.ts

```diff
@@ -2948,10 +2948,25 @@ namespace ts {
             return getClassExtendsHeritageClauseElement(<ClassLikeDeclaration>type.symbol.valueDeclaration);
         }
 
+        function getRequiredTypeArgumentCount(typeParameters: TypeParameter[]) {
+            if (typeParameters) {
+                for (let i = typeParameters.length - 1; i >= 0; i--) {
+                    if (!typeParameters[i].default) {

```

Comment:
This isn't ES3-compliant and we don't downlevel this as `typeParameters[i]["default"]`.